src/TufValidatedComposerRepository.php

```diff
@@ -88,9 +89,52 @@ protected function configurePackageTransportOptions(PackageInterface $package)
             'repository' => $config['url'],
             'target' => $package->getName() . '/' . $package->getVersion(),
         ];
+        if ($this->updater) {
+            try {
+                $options['max_file_size'] = $this->updater->getLength($options['tuf']['target']);
+            } catch (NotFoundException $e) {
+                $options['max_file_size'] = Updater::MAXIMUM_DOWNLOAD_BYTES;
+            }
+        }
         $package->setTransportOptions($options);
     }
 
+    /**
+     * Extracts a TUF target path from a full URL.
+     *
+     * @param string $url
+     *   A URL.
+     *
+     * @return string
+     *   A TUF target path derived from the URL.
+     */
+    private function getTargetFromUrl(string $url): string
+    {
+        $config = $this->getRepoConfig();
+        $target = str_replace($config['url'], null, $url);
+        return ltrim($target, '/');
+    }
+
+    /**
+     * Reacts before metadata is downloaded.
+     *
+     * @param PreFileDownloadEvent $event
+     *   The event object.
+     */
+    public function prepareMetadata(PreFileDownloadEvent $event): void
+    {
+        if ($this->updater) {
```

Comment:
Is the updater being set our flag for the repo being TUF protected? 

I am not sure this is very clear. When I first looked at this I thought maybe sometimes when the event was fired the updater hadn't be set up yet in some cycle.

So we replace all instances of `ComposerRepository` with an instance of an instance of `TufValidatedComposerRepository` correct? But depending on whether `$repoConfig['tuf']` is set some these instances will be protected by TUF and some won't. And depending on that `$this->updater)` will or won't be set?

Could we either 
1. set a flag in the constructor `$this->repoIsTufEnabled = isset($repoConfig['tuf'])` or 
2. create private method `function isTufEnabled() { return isset($this->updater) }: bool`

This way the 4 cases of `if ($this->updater) {` become self documenting

`if ($this->isTufEnabled()) {`

Reply:
Added a `isTufEnabled()` method.